pygin/pygin/src/panel.cpp

```diff
@@ -0,0 +1,61 @@
+#include "headers.hpp"
+
+#include "panel.h"
+#include "far_api.hpp"
+#include "py.integer.hpp"
+
+PluginPanelItem*
+get_find_data::get_items(py::object const& find_data, size_t& item_count, OPERATION_MODES& mode)
+{
+	find_data.ensure_type(far_api::type("GetFindDataInfo"sv));
+	auto py_items = py::cast<py::list>(find_data["PanelItems"sv]);
+	auto size = py_items.size();
+	std::unique_ptr<char[]> mem(new char[sizeof(get_find_data)
+		+ sizeof(PluginPanelItem) * size]);
+	auto* result = reinterpret_cast<PluginPanelItem*>(
+		mem.get() + sizeof(get_find_data));
+	new (mem.get()) get_find_data(py_items, result);
+	item_count = size;
+	mode = py::cast<OPERATION_MODES>(find_data["OpMode"sv]);
+	mem.release();
+	return result;
+}
+
+void get_find_data::free(PluginPanelItem* items)
+{
+	delete reinterpret_cast<get_find_data*>(
+		reinterpret_cast<char*>(items) - sizeof(get_find_data));
+}
+
+get_find_data::get_find_data(py::list const& py_items, PluginPanelItem* items)
+{
+	static const std::pair< wchar_t const* (PluginPanelItem::*), char const*> string_fields[] = {
+		{&PluginPanelItem::FileName, "FileName"},
+		{&PluginPanelItem::AlternateFileName, "AlternateFileName"},
+		{&PluginPanelItem::Description, "Description"},
+		{&PluginPanelItem::Owner, "Owner"},
+	};
+
+	this->strings.reserve(sizeof(string_fields) / sizeof(string_fields[0]) * py_items.size());
```

Comment:
std::size

Reply:
Hm... In fact I googled for it hoping they introduced this in the newer standards from `boost::size`... and missed somehow. I'll fix it.

Reply:
Fixed in 5ef1f612ff8787268b8d48658e3d1e8cad9b7272.